<!DOCTYPE html>
<html lang="en">

<head><script src="/workshop_nfRNAseq/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=workshop_nfRNAseq/livereload" data-no-instant defer></script>
  
  <meta charset="utf-8">



<meta name="viewport" content="width=device-width, initial-scale=1.0">


  
    <meta name="description" content="
https://nf-co.re/rnaseq/3.14.0
nf-core/RNAseq the most popular nf-core pipeline.

  

nf-core list -s stars
lets use nf-core tools to build a command to run the nf-core/RNAseq pipeline

  

mkdir ~/workshop/nfcore
cd ~/workshop/nfcore  

nf-core launch -h


  

nf-core launch nf-core/rnaseq

  

- use the arrow keys to select the following parameters
	- select release - version &#39;3.14.0&#39; 
selecting the release will download the pipeline, effectively running &rsquo;nextflow pull
you can see the key nextflow files are pulled to the ~/.nextflow directory, including the key executables and configs (nextflow.config, main.nf)
(exit with control&#43;c if needed)">
  


<meta name="color-scheme" content="light dark">







<meta name="generator" content="Hugo 0.135.0">
  <title>nfRNAseq | SAGG bioinformatics workshop</title>
  <link rel="canonical" href="http://localhost:1313/workshop_nfRNAseq/nfrnaseq/">




  








  
    
  
  
    <link rel="stylesheet" href="/workshop_nfRNAseq/css/base.css">

    <link rel="stylesheet" href="/workshop_nfRNAseq/css/syntax.css">



</head>

<body>
  <nav class="u-background">
  <div class="u-wrapper">
    <ul class="Banner">
      <li class="Banner-item Banner-item--title">
        <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/">SAGG bioinformatics workshop</a>
      </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/agenda/">agenda</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/setup_environment/">setup</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nextflow/">nextflow</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nf-core/">nf-core</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/">nfRNAseq</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/differentialabundance/">DifferentialAbundance</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/srna/">sRNA</a>
        </li>
      
    </ul>
  </div>
</nav>

  <main>
    <div class="u-wrapper">
      <div class="u-padding">
        

  <header class="Heading">
  <h1 class="Heading-title">
    <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/" rel="bookmark">nfRNAseq</a>
  </h1>
  
</header>
  <p><img src="..//nfRNAseq/nfRNAseq_pipeline.png" alt="nf-core/RNAseq"></p>
<p><a href="https://nf-co.re/rnaseq/3.14.0">https://nf-co.re/rnaseq/3.14.0</a></p>
<p>nf-core/RNAseq the most popular nf-core pipeline.</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">nf-core list -s stars</span></span></code></pre></div>
<p>lets use nf-core tools to build a command to run the nf-core/RNAseq pipeline</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir ~/workshop/nfcore
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> ~/workshop/nfcore  
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">nf-core launch -h</span></span></code></pre></div>
<p><img src="nfcorelaunch.png" alt="nf-core launch"></p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">nf-core launch nf-core/rnaseq</span></span></code></pre></div>

  

<pre tabindex="0"><code>- use the arrow keys to select the following parameters
	- select release - version &#39;3.14.0&#39; </code></pre>
<p>selecting the release will download the pipeline, effectively running &rsquo;nextflow pull
you can see the key nextflow files are pulled to the ~/.nextflow directory, including the key executables and configs (nextflow.config, main.nf)<br>
(exit with control+c if needed)</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">ls /home/workshop/.nextflow/assets/nf-core/rnaseq/</span></span></code></pre></div>
<p>This is one of many ways to pull a nf-core pipeline.</p>
<p>you can then select GUI or command line, with the GUI giving a URL to access via a browser by copying and pasting.</p>
<p><img src="GUI.png" alt=""></p>
<p>let&rsquo;s continue with adding variables with nf-core launch</p>

  

<pre tabindex="0"><code></code></pre>
<p>use the arrow keys to select the parameters to build a &lsquo;run&rsquo; command with the selected parameters
&lsquo;Web based&rsquo;</p>
<ul>
<li>LOTS of TEXT to FILL OUT here</li>
<li>if we have time at the end can go through other pipelines in more details if anyone has specific questions</li>
</ul>

  

<pre tabindex="0"><code>- you can exit the GUI (graphical user interphase) using **control** + **c**



	1. Merge re-sequenced FastQ files (cat)
	2. Sub-sample FastQ files and auto-infer strandedness (fq, Salmon)
	3. Read QC (FastQC)
	4. UMI extraction (UMI-tools)
	5. Adapter and quality trimming (Trim Galore!)
	6. Removal of genome contaminants (BBSplit)
	7. Removal of ribosomal RNA (SortMeRNA)
	8. Choice of multiple alignment and quantification routes:
		1. STAR -&gt; Salmon
		2. STAR -&gt; RSEM
		3. HiSAT2 -&gt; NO QUANTIFICATION
	9. Sort and index alignments (SAMtools)
	10. UMI-based deduplication (UMI-tools)
	11. Duplicate read marking (picard MarkDuplicates)
	12. Transcript assembly and quantification (StringTie)
	13. Create bigWig coverage files (BEDTools, bedGraphToBigWig)
	14. Extensive quality control:
		1. RSeQC
		2. Qualimap
		3. dupRadar
		4. Preseq
		5. DESeq2
	15. Pseudoalignment and quantification (Salmon or ‘Kallisto’; optional)
	16. Present QC for raw read, alignment, gene biotype, sample similarity, and strand-specificity checks (MultiQC, R)

In this section we will work through setting up the nf-core/RNAseq pipeline. We will be chosing specific parameters and finally, going through was the output looks like.

# setting up nf-core/RNAseq

#### dataset
The dataset used throughout this workshop is as follow:</code></pre>
<ul>
<li>16 Samples sequenced with an MGI400 sequencer at SAGC using the Tecan Universal RNA-seq library protocol.</li>
<li>2 different cancer cell lines (human)</li>
<li>treatment vs control</li>
<li>4 replicates for each</li>
</ul>

  

<pre tabindex="0"><code>***NOTE*** some concessions had to be made to work with this for workshop. Taking into account the large file sizes, the long run times and need for high compute resources.

#### what are the inputs to an RNAseq pipeline

# .fastq /.fastq.gz</code></pre>

  

<pre tabindex="0"><code>![](../fastqfile.png)

# .fasta

# .gtf

# .bed
![](../bedfile.png)

#### Genomic file formats
- now that we have gone through how to run the nf-core/RNAseq pipeline. Let&#39;s look at the inputs and outputs in detail.
- using this chance to describe what are the key genomic file formats used in RNAseq and beyond
- Knowing what these files are isn&#39;t only important in finding which files to use for a pipeline, but a key foundation of genomic bioinformati$
- being able to use and manipulate each file open&#39;s up many opportunities, and is often required for troubleshooting wehn something has gone w$
- many files are plain text files, this means they can be manipluated with basic text editing. I&#39;ll be going through some examples.
- For a more visual perspective, we&#39;ll also be using a genome browser, IGV (Integrative Genome Browser) to get a feel for what information eac$


# bam
![](../bamfile.png)

# bigwig



[link to multiqc report](../multiqc_report.html)</code></pre>

  

  

  





  
  



      </div>
    </div>
  </main>
  
  <footer class="Footer">
    <div class="u-wrapper">
      <div class="u-padding u-noboosting">
        2024 Daniel Thomson &#32; <a href="https://sa-genomics.com.au/our-team" rel="license">SA Genomics Centre</a>
      </div>
    </div>
  </footer>

</body>

</html>
