<!DOCTYPE html>
<html lang="en">

<head><script src="/workshop_nfRNAseq/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=workshop_nfRNAseq/livereload" data-no-instant defer></script>
  
  <meta charset="utf-8">



<meta name="viewport" content="width=device-width, initial-scale=1.0">


  
    <meta name="description" content="
https://nf-co.re/rnaseq/3.14.0
nf-core/RNAseq the most popular nf-core pipeline.

  

nf-core list -s stars
lets use nf-core tools to build a command to run the nf-core/RNAseq pipeline

  

mkdir -p  ~/workshop/nfRNAseq
cd ~/workshop/nfRNAseq 

nf-core launch -h
Launch a pipeline using a web GUI or command line prompts.                                      
Uses the pipeline schema file to collect inputs for all available pipeline parameters.
When finished, saves a file with the selected parameters which can be passed to Nextflow using the
-params-file option.">
  


<meta name="color-scheme" content="light dark">







<meta name="generator" content="Hugo 0.135.0">
  <title>nfRNAseq | SAGG bioinformatics workshop</title>
  <link rel="canonical" href="http://localhost:1313/workshop_nfRNAseq/nfrnaseq/">




  








  
    
  
  
    <link rel="stylesheet" href="/workshop_nfRNAseq/css/base.css">

    <link rel="stylesheet" href="/workshop_nfRNAseq/css/syntax.css">



</head>

<body>
  <nav class="u-background">
  <div class="u-wrapper">
    <ul class="Banner">
      <li class="Banner-item Banner-item--title">
        <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/">SAGG bioinformatics workshop</a>
      </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/agenda/">agenda</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/setup_environment/">setup</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nextflow/">nextflow</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nf-core/">nf-core</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/">nfRNAseq</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/differentialabundance/">DifferentialAbundance</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/srna/">sRNA</a>
        </li>
      
    </ul>
  </div>
</nav>

  <main>
    <div class="u-wrapper">
      <div class="u-padding">
        

  <header class="Heading">
  <h1 class="Heading-title">
    <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/" rel="bookmark">nfRNAseq</a>
  </h1>
  
</header>
  <p><img src="..//nfRNAseq/nfRNAseq_pipeline.png" alt="nf-core/RNAseq"></p>
<p><a href="https://nf-co.re/rnaseq/3.14.0">https://nf-co.re/rnaseq/3.14.0</a></p>
<p>nf-core/RNAseq the most popular nf-core pipeline.</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">nf-core list -s stars</span></span></code></pre></div>
<p>lets use nf-core tools to build a command to run the nf-core/RNAseq pipeline</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir -p  ~/workshop/nfRNAseq
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> ~/workshop/nfRNAseq 
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">nf-core launch -h</span></span></code></pre></div>
<p>Launch a pipeline using a web GUI or command line prompts.                                      <br>
Uses the pipeline schema file to collect inputs for all available pipeline parameters.
When finished, saves a file with the selected parameters which can be passed to Nextflow using the
-params-file option.</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">nf-core launch nf-core/rnaseq</span></span></code></pre></div>

  

<pre tabindex="0"><code>- use the arrow keys to select the following parameters
	- select release - version &#39;3.14.0&#39; </code></pre>
<p>selecting the release will download the pipeline, effectively running &rsquo;nextflow pull
you can see the key nextflow files are pulled to the ~/.nextflow directory, including the key executables and configs (nextflow.config, main.nf)<br>
(exit with control+c if needed)</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">ls /home/workshop/.nextflow/assets/nf-core/rnaseq/</span></span></code></pre></div>
<p>This is one of many ways to pull a nf-core pipeline.</p>
<p>you can then select GUI or command line, with the GUI giving a URL to access via a browser by copying and pasting.</p>
<p><img src="GUI.png" alt=""></p>
<p>we can continue with adding variables with nf-core launch</p>

  

<pre tabindex="0"><code>	-name test
	-profile singularity \
	-resume
	--input nfSampleSheet.csv \
	--outdir Out \
	--fasta Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa \
	--gtf Homo_sapiens.GRCh38.111.gtf \</code></pre>
<p>this will give you a nf-core launch command with those variable</p>

  

<pre tabindex="0"><code>nf-core launch --id 1728482936_6064841c2138</code></pre>
<ul>
<li>
<p>alternatively you can use a nextflow run command</p>
</li>
<li>
<p>no need to run this command, in the interest of time (and the lack of disk space on your intance), I&rsquo;ve pre-prepared the outputs for this run</p>
</li>
</ul>
<p>navigate to run directory to see the nextflow run command</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">cd</span> ~/workshop/RNAseq
</span></span><span class="line"><span class="cl">ls
</span></span><span class="line"><span class="cl">cat nf_rnaseq.sh</span></span></code></pre></div>
<p><img src="runcommand.png" alt=""></p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">cat nfSampleSheet.csv</span></span></code></pre></div>
<p><img src="samplesheet.png" alt=""></p>



<h4 id="dataset">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#dataset">dataset</a>
</h4>
<p>The dataset used throughout this workshop is as follow:</p>

  

<pre tabindex="0"><code>- 16 Samples sequenced with an MGI400 sequencer at SAGC using the Tecan Universal RNA-seq library protocol.
- 2 different cancer cell lines (human)
- treatment vs control
- 4 replicates for each</code></pre>
<ul>
<li>most of this information is reflected in the samplesheet and run command</li>
</ul>



<h4 id="run-summary">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#run-summary">Run Summary</a>
</h4>
<p>list results directory. All nf-core outputs have a consistant structure of the outputs</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">tree /home/workshop/workshop/nfRNAseq/outs</span></span></code></pre></div>
<p><a href="../execution_timeline_2024-10-05_16-02-39.html">link to execution timeline</a></p>
<p><a href="../execution_report_2024-10-05_16-02-39.html">link to execution report</a></p>



<h1 id="genomics-files-background">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#genomics-files-background">Genomics Files Background</a>
</h1>
<p>Quick background on genomic file formats to help describe the inputs and outputs of an NGS experiment.
Knowing what these files are isn&rsquo;t only important in finding which files to use for a pipeline, but a key foundation of genomic bioinformatics
Being able to use and manipulate each file open&rsquo;s up many opportunities, and is often required for troubleshooting                        <br>
Many are plain text files, this means they can be manipluated with basic text editing.</p>



<h5 id="fastq-fastqgz">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#fastq-fastqgz">.fastq /.fastq.gz</a>
</h5>
<p><img src="../fastqfile.png" alt=""></p>



<h4 id="fasta">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#fasta">.fasta</a>
</h4>
<p>the genome.fa file is a plain text representation of the genome sequence. This is the &lsquo;reference&rsquo; to which the sequencing files (.fastq) are alligned.</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">head ./smallRNA/outs/mirtrace/<span class="o">[</span>:<span class="o">]</span>/mirtrace/qc_passed_reads.rnatype_unknown.collapsed/Acontrol1.fastp.fasta</span></span></code></pre></div>



<h4 id="gtf">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#gtf">.gtf</a>
</h4>
<p>genes.gtf ; a genomic interval file referencing the genome. This file depicts the genes/ transcripts. It&rsquo;s a required for counting where reads are mapped to, and often has alot more annotation data in regards to the gene/transcript.</p>



<h4 id="bed">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#bed">.bed</a>
</h4>
<p><img src="../bedfile.png" alt=""></p>



<h4 id="bam">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#bam">bam</a>
</h4>
<p><img src="../bamfile.png" alt=""></p>



<h1 id="walkthough-of-outputs-nf-corernaseq">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#walkthough-of-outputs-nf-corernaseq">Walkthough of outputs nf-core/RNAseq</a>
</h1>
<p>the multiqc summary is a real strength of nf-core pipeline.
alot of the key analyses are captured</p>
<p><a href="../multiqc_report.html">link to multiqc report</a></p>

  

<pre tabindex="0"><code>Overview of the key processes run with nf-core/RNAseq.
- identifying steps to trouble shooting the success of a run.
- far more than just mapping (star) and counting (RSEM)
- demonstrating the need for a workflow manager</code></pre>



<h3 id="preprocessing">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#preprocessing">Preprocessing</a>
</h3>
<ol>
<li>cat - Merge re-sequenced FastQ files</li>
<li>FastQC - Raw read QC</li>
</ol>
<p><a href="../Acontrol1_1_fastqc.html">link to fastqc.html</a></p>
<ol start="3">
<li>UMI-tools extract - UMI barcode extraction</li>
<li>TrimGalore - Adapter and quality trimming</li>
<li>BBSplit - Removal of genome contaminants</li>
<li>SortMeRNA - Removal of ribosomal RNA</li>
</ol>



<h3 id="alignment-and-quantification">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#alignment-and-quantification">Alignment and quantification</a>
</h3>
<ol>
<li>STAR and Salmon - Fast spliced aware genome alignment and transcriptome quantification</li>
<li>STAR via RSEM - Alignment and quantification of expression levels</li>
<li>HISAT2 - Memory efficient splice aware alignment to a reference</li>
</ol>



<h3 id="alignment-post-processing">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#alignment-post-processing">Alignment post-processing</a>
</h3>
<ol>
<li>SAMtools - Sort and index alignments</li>
<li>UMI-tools dedup - UMI-based deduplication</li>
<li>picard MarkDuplicates - Duplicate read marking</li>
</ol>



<h3 id="other-steps">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#other-steps">Other steps</a>
</h3>
<ol>
<li>StringTie - Transcript assembly and quantification</li>
<li>BEDTools and bedGraphToBigWig - Create bigWig coverage files</li>
</ol>



<h3 id="quality-control">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#quality-control">Quality control</a>
</h3>
<ol>
<li>RSeQC - Various RNA-seq QC metrics</li>
<li>Qualimap - Various RNA-seq QC metrics</li>
<li>dupRadar - Assessment of technical / biological read duplication</li>
<li>Preseq - Estimation of library complexity</li>
<li>featureCounts - Read counting relative to gene biotype</li>
<li>DESeq2 - PCA plot and sample pairwise distance heatmap and dendrogram
7 MultiQC - Present QC for raw reads, alignment, read counting and sample similiarity</li>
</ol>



<h3 id="pseudoalignment-and-quantification">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#pseudoalignment-and-quantification">Pseudoalignment and quantification</a>
</h3>
<ol>
<li>Salmon - Wicked fast gene and isoform quantification relative to the transcriptome</li>
<li>Kallisto - Near-optimal probabilistic RNA-seq quantification</li>
<li>Workflow reporting and genomes</li>
<li>Reference genome files - Saving reference genome indices/files</li>
<li>Pipeline information - Report metrics generated during the workflow execution</li>
</ol>
<p><a href="../execution_report_2024-10-05_16-02-39.html">link to execution report</a></p>

  

  

  





  
  



      </div>
    </div>
  </main>
  
  <footer class="Footer">
    <div class="u-wrapper">
      <div class="u-padding u-noboosting">
        2024 Daniel Thomson &#32; <a href="https://sa-genomics.com.au/our-team" rel="license">SA Genomics Centre</a>
      </div>
    </div>
  </footer>

</body>

</html>
