<!DOCTYPE html>
<html lang="en">

<head><script src="/workshop_nfRNAseq/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=workshop_nfRNAseq/livereload" data-no-instant defer></script>
  
  <meta charset="utf-8">



<meta name="viewport" content="width=device-width, initial-scale=1.0">


  
    <meta name="description" content="
https://nf-co.re/rnaseq/3.14.0
nf-core/RNAseq the most popular nf-core pipeline.

  

nf-core list -s stars
lets use nf-core tools to build a command to run the nf-core/RNAseq pipeline

  

mkdir ~/workshop/nfcore
cd ~/workshop/nfcore  

nf-core launch -h
Launch a pipeline using a web GUI or command line prompts.                                      
Uses the pipeline schema file to collect inputs for all available pipeline parameters.
When finished, saves a file with the selected parameters which can be passed to Nextflow using the
-params-file option.">
  


<meta name="color-scheme" content="light dark">







<meta name="generator" content="Hugo 0.135.0">
  <title>nfRNAseq | SAGG bioinformatics workshop</title>
  <link rel="canonical" href="http://localhost:1313/workshop_nfRNAseq/nfrnaseq/">




  








  
    
  
  
    <link rel="stylesheet" href="/workshop_nfRNAseq/css/base.css">

    <link rel="stylesheet" href="/workshop_nfRNAseq/css/syntax.css">



</head>

<body>
  <nav class="u-background">
  <div class="u-wrapper">
    <ul class="Banner">
      <li class="Banner-item Banner-item--title">
        <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/">SAGG bioinformatics workshop</a>
      </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/agenda/">agenda</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/setup_environment/">setup</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nextflow/">nextflow</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nf-core/">nf-core</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/">nfRNAseq</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/differentialabundance/">DifferentialAbundance</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/srna/">sRNA</a>
        </li>
      
    </ul>
  </div>
</nav>

  <main>
    <div class="u-wrapper">
      <div class="u-padding">
        

  <header class="Heading">
  <h1 class="Heading-title">
    <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/" rel="bookmark">nfRNAseq</a>
  </h1>
  
</header>
  <p><img src="..//nfRNAseq/nfRNAseq_pipeline.png" alt="nf-core/RNAseq"></p>
<p><a href="https://nf-co.re/rnaseq/3.14.0">https://nf-co.re/rnaseq/3.14.0</a></p>
<p>nf-core/RNAseq the most popular nf-core pipeline.</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">nf-core list -s stars</span></span></code></pre></div>
<p>lets use nf-core tools to build a command to run the nf-core/RNAseq pipeline</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir ~/workshop/nfcore
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> ~/workshop/nfcore  
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">nf-core launch -h</span></span></code></pre></div>
<p>Launch a pipeline using a web GUI or command line prompts.                                      <br>
Uses the pipeline schema file to collect inputs for all available pipeline parameters.
When finished, saves a file with the selected parameters which can be passed to Nextflow using the
-params-file option.</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">nf-core launch nf-core/rnaseq</span></span></code></pre></div>

  

<pre tabindex="0"><code>- use the arrow keys to select the following parameters
	- select release - version &#39;3.14.0&#39; </code></pre>
<p>selecting the release will download the pipeline, effectively running &rsquo;nextflow pull
you can see the key nextflow files are pulled to the ~/.nextflow directory, including the key executables and configs (nextflow.config, main.nf)<br>
(exit with control+c if needed)</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">ls /home/workshop/.nextflow/assets/nf-core/rnaseq/</span></span></code></pre></div>
<p>This is one of many ways to pull a nf-core pipeline.</p>
<p>you can then select GUI or command line, with the GUI giving a URL to access via a browser by copying and pasting.</p>
<p><img src="GUI.png" alt=""></p>
<p>we can continue with adding variables with nf-core launch</p>

  

<pre tabindex="0"><code>	-name test
	-profile singularity \
	-resume
	--input nfSampleSheet.csv \
	--outdir Out \
	--fasta Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa \
	--gtf Homo_sapiens.GRCh38.111.gtf \</code></pre>
<p>this will give you a nf-core launch command with those variable</p>

  

<pre tabindex="0"><code>nf-core launch --id 1728482936_6064841c2138</code></pre>
<ul>
<li>
<p>alternatively you can use a nextflow run command</p>
</li>
<li>
<p>no need to run this command, in the interest of time (and the lack of disk space on your intance), I&rsquo;ve pre-prepared the outputs for this run. We will run the next pipeline to completion.</p>
</li>
</ul>
<p>navigate to run directory to see the nextflow run command</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">cd</span> ~/workshop/RNAseq
</span></span><span class="line"><span class="cl">ls
</span></span><span class="line"><span class="cl">cat nf_rnaseq.sh</span></span></code></pre></div>
<p><img src="runcommand.png" alt=""></p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">cat nfSampleSheet.csv</span></span></code></pre></div>
<p><img src="samplesheet.png" alt=""></p>



<h4 id="dataset">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#dataset">dataset</a>
</h4>
<p>The dataset used throughout this workshop is as follow:</p>

  

<pre tabindex="0"><code>- 16 Samples sequenced with an MGI400 sequencer at SAGC using the Tecan Universal RNA-seq library protocol.
- 2 different cancer cell lines (human)
- treatment vs control
- 4 replicates for each</code></pre>
<ul>
<li>the information is reflected in the samplesheet and run command</li>
</ul>



<h1 id="results">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#results">Results</a>
</h1>
<pre><code>1. Merge re-sequenced FastQ files (cat)
2. Sub-sample FastQ files and auto-infer strandedness (fq, Salmon)
3. Read QC (FastQC)
4. UMI extraction (UMI-tools)
5. Adapter and quality trimming (Trim Galore!)
6. Removal of genome contaminants (BBSplit)
7. Removal of ribosomal RNA (SortMeRNA)
8. Choice of multiple alignment and quantification routes:
	1. STAR -&gt; Salmon
	2. STAR -&gt; RSEM
	3. HiSAT2 -&gt; NO QUANTIFICATION
9. Sort and index alignments (SAMtools)
10. UMI-based deduplication (UMI-tools)
11. Duplicate read marking (picard MarkDuplicates)
12. Transcript assembly and quantification (StringTie)
13. Create bigWig coverage files (BEDTools, bedGraphToBigWig)
14. Extensive quality control:
	1. RSeQC
	2. Qualimap
	3. dupRadar
	4. Preseq
	5. DESeq2
15. Pseudoalignment and quantification (Salmon or ‘Kallisto’; optional)
16. Present QC for raw read, alignment, gene biotype, sample similarity, and strand-specificity checks (MultiQC, R)
</code></pre>
<p>In this section we will work through setting up the nf-core/RNAseq pipeline. We will be chosing specific parameters and finally, going through was the output looks like.</p>



<h1 id="setting-up-nf-corernaseq">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#setting-up-nf-corernaseq">setting up nf-core/RNAseq</a>
</h1>



<h4 id="dataset-1">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#dataset-1">dataset</a>
</h4>
<p>The dataset used throughout this workshop is as follow:</p>

  

<pre tabindex="0"><code>- 16 Samples sequenced with an MGI400 sequencer at SAGC using the Tecan Universal RNA-seq library protocol.
- 2 different cancer cell lines (human)
- treatment vs control
- 4 replicates for each</code></pre>
<p><em><strong>NOTE</strong></em> some concessions had to be made to work with this for workshop. Taking into account the large file sizes, the long run times and need for high compute resources.</p>



<h4 id="what-are-the-inputs-to-an-rnaseq-pipeline">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#what-are-the-inputs-to-an-rnaseq-pipeline">what are the inputs to an RNAseq pipeline</a>
</h4>



<h1 id="fastq-fastqgz">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#fastq-fastqgz">.fastq /.fastq.gz</a>
</h1>

  

<pre tabindex="0"><code></code></pre>
<p><img src="../fastqfile.png" alt=""></p>



<h1 id="fasta">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#fasta">.fasta</a>
</h1>



<h1 id="gtf">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#gtf">.gtf</a>
</h1>



<h1 id="bed">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#bed">.bed</a>
</h1>
<p><img src="../bedfile.png" alt=""></p>



<h4 id="genomic-file-formats">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#genomic-file-formats">Genomic file formats</a>
</h4>
<ul>
<li>now that we have gone through how to run the nf-core/RNAseq pipeline. Let&rsquo;s look at the inputs and outputs in detail.</li>
<li>using this chance to describe what are the key genomic file formats used in RNAseq and beyond</li>
<li>Knowing what these files are isn&rsquo;t only important in finding which files to use for a pipeline, but a key foundation of genomic bioinformati$</li>
<li>being able to use and manipulate each file open&rsquo;s up many opportunities, and is often required for troubleshooting wehn something has gone w$</li>
<li>many files are plain text files, this means they can be manipluated with basic text editing. I&rsquo;ll be going through some examples.</li>
<li>For a more visual perspective, we&rsquo;ll also be using a genome browser, IGV (Integrative Genome Browser) to get a feel for what information eac$</li>
</ul>



<h1 id="bam">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#bam">bam</a>
</h1>
<p><img src="../bamfile.png" alt=""></p>



<h1 id="bigwig">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#bigwig">bigwig</a>
</h1>
<p><a href="../multiqc_report.html">link to multiqc report</a></p>

  

  

  





  
  



      </div>
    </div>
  </main>
  
  <footer class="Footer">
    <div class="u-wrapper">
      <div class="u-padding u-noboosting">
        2024 Daniel Thomson &#32; <a href="https://sa-genomics.com.au/our-team" rel="license">SA Genomics Centre</a>
      </div>
    </div>
  </footer>

</body>

</html>
