<!DOCTYPE html>
<html lang="en">

<head><script src="/workshop_nfRNAseq/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=workshop_nfRNAseq/livereload" data-no-instant defer></script>
  
  <meta charset="utf-8">



<meta name="viewport" content="width=device-width, initial-scale=1.0">


  
    <meta name="description" content="
https://nf-co.re/rnaseq/3.14.0
nf-core/RNA the most popular nf-core pipeline.
1. Merge re-sequenced FastQ files (cat)
2. Sub-sample FastQ files and auto-infer strandedness (fq, Salmon)
3. Read QC (FastQC)
4. UMI extraction (UMI-tools)
5. Adapter and quality trimming (Trim Galore!)
6. Removal of genome contaminants (BBSplit)
7. Removal of ribosomal RNA (SortMeRNA)
8. Choice of multiple alignment and quantification routes:
	1. STAR -&gt; Salmon
	2. STAR -&gt; RSEM
	3. HiSAT2 -&gt; NO QUANTIFICATION
9. Sort and index alignments (SAMtools)
10. UMI-based deduplication (UMI-tools)
11. Duplicate read marking (picard MarkDuplicates)
12. Transcript assembly and quantification (StringTie)
13. Create bigWig coverage files (BEDTools, bedGraphToBigWig)
14. Extensive quality control:
	1. RSeQC
	2. Qualimap
	3. dupRadar
	4. Preseq
	5. DESeq2
15. Pseudoalignment and quantification (Salmon or ‘Kallisto’; optional)
16. Present QC for raw read, alignment, gene biotype, sample similarity, and strand-specificity checks (MultiQC, R)

In this section we will work through setting up the nf-core/RNAseq pipeline. We will be chosing specific parameters and finally, going through was the output looks like.">
  


<meta name="color-scheme" content="light dark">







<meta name="generator" content="Hugo 0.135.0">
  <title>nfRNAseq | SAGG bioinformatics workshop</title>
  <link rel="canonical" href="http://localhost:1313/workshop_nfRNAseq/nfrnaseq/">




  








  
    
  
  
    <link rel="stylesheet" href="/workshop_nfRNAseq/css/base.css">

    <link rel="stylesheet" href="/workshop_nfRNAseq/css/syntax.css">



</head>

<body>
  <nav class="u-background">
  <div class="u-wrapper">
    <ul class="Banner">
      <li class="Banner-item Banner-item--title">
        <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/">SAGG bioinformatics workshop</a>
      </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/agenda/">agenda</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/setup_environment/">setup</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nextflow/">nextflow</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nf-core/">nf-core</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/">nfRNAseq</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/differentialabundance/">DifferentialAbundance</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/srna/">sRNA</a>
        </li>
      
    </ul>
  </div>
</nav>

  <main>
    <div class="u-wrapper">
      <div class="u-padding">
        

  <header class="Heading">
  <h1 class="Heading-title">
    <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/" rel="bookmark">nfRNAseq</a>
  </h1>
  
</header>
  <p><img src="/nfRNAseq/nfRNAseq_pipeline.png" alt="nf-core/RNAseq"></p>
<p><a href="https://nf-co.re/rnaseq/3.14.0">https://nf-co.re/rnaseq/3.14.0</a></p>
<p>nf-core/RNA the most popular nf-core pipeline.</p>
<pre><code>1. Merge re-sequenced FastQ files (cat)
2. Sub-sample FastQ files and auto-infer strandedness (fq, Salmon)
3. Read QC (FastQC)
4. UMI extraction (UMI-tools)
5. Adapter and quality trimming (Trim Galore!)
6. Removal of genome contaminants (BBSplit)
7. Removal of ribosomal RNA (SortMeRNA)
8. Choice of multiple alignment and quantification routes:
	1. STAR -&gt; Salmon
	2. STAR -&gt; RSEM
	3. HiSAT2 -&gt; NO QUANTIFICATION
9. Sort and index alignments (SAMtools)
10. UMI-based deduplication (UMI-tools)
11. Duplicate read marking (picard MarkDuplicates)
12. Transcript assembly and quantification (StringTie)
13. Create bigWig coverage files (BEDTools, bedGraphToBigWig)
14. Extensive quality control:
	1. RSeQC
	2. Qualimap
	3. dupRadar
	4. Preseq
	5. DESeq2
15. Pseudoalignment and quantification (Salmon or ‘Kallisto’; optional)
16. Present QC for raw read, alignment, gene biotype, sample similarity, and strand-specificity checks (MultiQC, R)
</code></pre>
<p>In this section we will work through setting up the nf-core/RNAseq pipeline. We will be chosing specific parameters and finally, going through was the output looks like.</p>



<h1 id="setting-up-nf-corernaseq">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#setting-up-nf-corernaseq">setting up nf-core/RNAseq</a>
</h1>



<h4 id="dataset">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#dataset">dataset</a>
</h4>
<p>The dataset used throughout this workshop is as follow:</p>

  

<pre tabindex="0"><code>- 16 Samples sequenced with an MGI400 sequencer at SAGC using the Tecan Universal RNA-seq library protocol.
- 2 different cancer cell lines (human)
- treatment vs control
- 4 replicates for each</code></pre>
<p><em><strong>NOTE</strong></em> some concessions had to be made to work with this for workshop. Taking into account the large file sizes, the long run times and need for high compute resources.</p>
<p>let&rsquo;s start by having</p>



<h4 id="genomic-file-formats">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#genomic-file-formats">Genomic file formats</a>
</h4>
<ul>
<li>now that we have gone through how to run the nf-core/RNAseq pipeline. Let&rsquo;s look at the inputs and outputs in detail.</li>
<li>using this chance to describe what are the key genomic file formats used in RNAseq and beyond</li>
<li>Knowing what these files are isn&rsquo;t only important in finding which files to use for a pipeline, but a key foundation of genomic bioinformatics.</li>
<li>being able to use and manipulate each file open&rsquo;s up many opportunities, and is often required for troubleshooting wehn something has gone wrong</li>
<li>many files are plain text files, this means they can be manipluated with basic text editing. I&rsquo;ll be going through some examples.</li>
<li>For a more visual perspective, we&rsquo;ll also be using a genome browser, IGV (Integrative Genome Browser) to get a feel for what information each of these files contain.</li>
</ul>



<h1 id="fastq">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#fastq">fastq</a>
</h1>

  

<pre tabindex="0"><code></code></pre>



<h1 id="fasta">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#fasta">fasta</a>
</h1>



<h1 id="gtf">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#gtf">gtf</a>
</h1>



<h1 id="bam">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#bam">bam</a>
</h1>



<h1 id="bigwig">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/#bigwig">bigwig</a>
</h1>

  

  

  





  
  



      </div>
    </div>
  </main>
  
  <footer class="Footer">
    <div class="u-wrapper">
      <div class="u-padding u-noboosting">
        2024 Daniel Thomson &#32; <a href="https://sa-genomics.com.au/our-team" rel="license">SA Genomics Centre</a>
      </div>
    </div>
  </footer>

</body>

</html>
