<!DOCTYPE html>
<html lang="en">

<head>
  
  <meta charset="utf-8">



<meta name="viewport" content="width=device-width, initial-scale=1.0">


  
    <meta name="description" content="https://nf-co.re/smrnaseq/2.2.1/




  nf-core/smRNAseq

Pipeline summary

Raw read QC (FastQC)
Adapter trimming (Trim Galore!)

Insert Size calculation
Collapse reads (seqcluster)


Contamination filtering (Bowtie2)
Alignment against miRBase mature miRNA (Bowtie1)
Alignment against miRBase hairpin

Unaligned reads from step 3 (Bowtie1)
Collapsed reads from step 2.2 (Bowtie1)


Post-alignment processing of miRBase hairpin

Basic statistics from step 3 and step 4.1 (SAMtools)
Analysis on miRBase, or MirGeneDB hairpin counts (edgeR)

TMM normalization and a table of top expression hairpin
MDS plot clustering samples
Heatmap of sample similarities


miRNA and isomiR annotation from step 4.1 (mirtop)


Alignment against host reference genome (Bowtie1)

Post-alignment processing of alignment against host reference genome (SAMtools)


Novel miRNAs and known miRNAs discovery (MiRDeep2)

Mapping against reference genome with the mapper module
Known and novel miRNA discovery with the mirdeep2 module


miRNA quality control (mirtrace)
Present QC for raw read, alignment, and expression results (MultiQC)





  running nf-core/sRNAseq

">
  


<meta name="color-scheme" content="light dark">







<meta name="generator" content="Hugo 0.135.0">
  <title>nf-core/smRNAseq | SAGG bioinformatics workshop</title>
  <link rel="canonical" href="https://sagc-bioinformatics.github.io/workshop_nfRNAseq/srna/">




  








  
    
  
  
      <link rel="stylesheet" href="/workshop_nfRNAseq/css/base.min.ca70ae019821d0d70b4488a7cda84c279a6ed672a4fe790a33e5b11ea4fe2e5b.css" integrity="sha256-ynCuAZgh0NcLRIinzahMJ5pu1nKk/nkKM&#43;WxHqT&#43;Lls=" crossorigin="anonymous">



</head>

<body>
  <nav class="u-background">
  <div class="u-wrapper">
    <ul class="Banner">
      <li class="Banner-item Banner-item--title">
        <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/">SAGG bioinformatics workshop</a>
      </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/agenda/">agenda</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/setup_environment/">setup</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nextflow/">nextflow</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nf-core/">nf-core</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/">nfRNAseq</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/differentialabundance/">DifferentialAbundance</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/srna/">sRNA</a>
        </li>
      
    </ul>
  </div>
</nav>

  <main>
    <div class="u-wrapper">
      <div class="u-padding">
        

  <header class="Heading">
  <h1 class="Heading-title">
    <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/srna/" rel="bookmark">nf-core/smRNAseq</a>
  </h1>
  
</header>
  <p><a href="https://nf-co.re/smrnaseq/2.2.1/">https://nf-co.re/smrnaseq/2.2.1/</a></p>



<h4 id="nf-coresmrnaseq">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/srna/#nf-coresmrnaseq">nf-core/smRNAseq</a>
</h4>
<p>Pipeline summary</p>
<ol>
<li>Raw read QC (FastQC)</li>
<li>Adapter trimming (Trim Galore!)
<ol>
<li>Insert Size calculation</li>
<li>Collapse reads (seqcluster)</li>
</ol>
</li>
<li>Contamination filtering (Bowtie2)</li>
<li>Alignment against miRBase mature miRNA (Bowtie1)</li>
<li>Alignment against miRBase hairpin
<ol>
<li>Unaligned reads from step 3 (Bowtie1)</li>
<li>Collapsed reads from step 2.2 (Bowtie1)</li>
</ol>
</li>
<li>Post-alignment processing of miRBase hairpin
<ol>
<li>Basic statistics from step 3 and step 4.1 (SAMtools)</li>
<li>Analysis on miRBase, or MirGeneDB hairpin counts (edgeR)
<ul>
<li>TMM normalization and a table of top expression hairpin</li>
<li>MDS plot clustering samples</li>
<li>Heatmap of sample similarities</li>
</ul>
</li>
<li>miRNA and isomiR annotation from step 4.1 (mirtop)</li>
</ol>
</li>
<li>Alignment against host reference genome (Bowtie1)
<ol>
<li>Post-alignment processing of alignment against host reference genome (SAMtools)</li>
</ol>
</li>
<li>Novel miRNAs and known miRNAs discovery (MiRDeep2)
<ol>
<li>Mapping against reference genome with the mapper module</li>
<li>Known and novel miRNA discovery with the mirdeep2 module</li>
</ol>
</li>
<li>miRNA quality control (mirtrace)</li>
<li>Present QC for raw read, alignment, and expression results (MultiQC)</li>
</ol>



<h4 id="running-nf-coresrnaseq">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/srna/#running-nf-coresrnaseq">running nf-core/sRNAseq</a>
</h4>
<p><img src="/sRNA/smRNAseq_log.png" alt=""></p>

  

  

  





  
  



      </div>
    </div>
  </main>
  
  <footer class="Footer">
    <div class="u-wrapper">
      <div class="u-padding u-noboosting">
        2024 Daniel Thomson &#32; <a href="https://sa-genomics.com.au/our-team" rel="license">SA Genomics Centre</a>
      </div>
    </div>
  </footer>

</body>

</html>
