<!DOCTYPE html>
<html lang="en">

<head><script src="/workshop_nfRNAseq/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=workshop_nfRNAseq/livereload" data-no-instant defer></script>
  
  <meta charset="utf-8">



<meta name="viewport" content="width=device-width, initial-scale=1.0">


  
    <meta name="description" content="https://nf-co.re/smrnaseq/2.2.1/




  nf-core/smRNAseq

test
Pipeline summary

Raw read QC (FastQC)
Adapter trimming (Trim Galore!)

Insert Size calculation
Collapse reads (seqcluster)


Contamination filtering (Bowtie2)
Alignment against miRBase mature miRNA (Bowtie1)
Alignment against miRBase hairpin

Unaligned reads from step 3 (Bowtie1)
Collapsed reads from step 2.2 (Bowtie1)


Post-alignment processing of miRBase hairpin

Basic statistics from step 3 and step 4.1 (SAMtools)
Analysis on miRBase, or MirGeneDB hairpin counts (edgeR)

TMM normalization and a table of top expression hairpin
MDS plot clustering samples
Heatmap of sample similarities


miRNA and isomiR annotation from step 4.1 (mirtop)


Alignment against host reference genome (Bowtie1)

Post-alignment processing of alignment against host reference genome (SAMtools)


Novel miRNAs and known miRNAs discovery (MiRDeep2)

Mapping against reference genome with the mapper module
Known and novel miRNA discovery with the mirdeep2 module


miRNA quality control (mirtrace)
Present QC for raw read, alignment, and expression results (MultiQC)





  running nf-core/sRNAseq

">
  


<meta name="color-scheme" content="light dark">







<meta name="generator" content="Hugo 0.135.0">
  <title>nf-core/smRNAseq | SAGG bioinformatics workshop</title>
  <link rel="canonical" href="http://localhost:1313/workshop_nfRNAseq/srna/">




  








  
    
  
  
    <link rel="stylesheet" href="/workshop_nfRNAseq/css/base.css">

    <link rel="stylesheet" href="/workshop_nfRNAseq/css/syntax.css">



</head>

<body>
  <nav class="u-background">
  <div class="u-wrapper">
    <ul class="Banner">
      <li class="Banner-item Banner-item--title">
        <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/">SAGG bioinformatics workshop</a>
      </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/agenda/">agenda</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/setup_environment/">setup</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nextflow/">nextflow</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nf-core/">nf-core</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/">nfRNAseq</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/differentialabundance/">DifferentialAbundance</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/srna/">sRNA</a>
        </li>
      
    </ul>
  </div>
</nav>

  <main>
    <div class="u-wrapper">
      <div class="u-padding">
        

  <header class="Heading">
  <h1 class="Heading-title">
    <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/srna/" rel="bookmark">nf-core/smRNAseq</a>
  </h1>
  
</header>
  <p><a href="https://nf-co.re/smrnaseq/2.2.1/">https://nf-co.re/smrnaseq/2.2.1/</a></p>



<h4 id="nf-coresmrnaseq">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/srna/#nf-coresmrnaseq">nf-core/smRNAseq</a>
</h4>
<p>test</p>
<p>Pipeline summary</p>
<ol>
<li>Raw read QC (FastQC)</li>
<li>Adapter trimming (Trim Galore!)
<ol>
<li>Insert Size calculation</li>
<li>Collapse reads (seqcluster)</li>
</ol>
</li>
<li>Contamination filtering (Bowtie2)</li>
<li>Alignment against miRBase mature miRNA (Bowtie1)</li>
<li>Alignment against miRBase hairpin
<ol>
<li>Unaligned reads from step 3 (Bowtie1)</li>
<li>Collapsed reads from step 2.2 (Bowtie1)</li>
</ol>
</li>
<li>Post-alignment processing of miRBase hairpin
<ol>
<li>Basic statistics from step 3 and step 4.1 (SAMtools)</li>
<li>Analysis on miRBase, or MirGeneDB hairpin counts (edgeR)
<ul>
<li>TMM normalization and a table of top expression hairpin</li>
<li>MDS plot clustering samples</li>
<li>Heatmap of sample similarities</li>
</ul>
</li>
<li>miRNA and isomiR annotation from step 4.1 (mirtop)</li>
</ol>
</li>
<li>Alignment against host reference genome (Bowtie1)
<ol>
<li>Post-alignment processing of alignment against host reference genome (SAMtools)</li>
</ol>
</li>
<li>Novel miRNAs and known miRNAs discovery (MiRDeep2)
<ol>
<li>Mapping against reference genome with the mapper module</li>
<li>Known and novel miRNA discovery with the mirdeep2 module</li>
</ol>
</li>
<li>miRNA quality control (mirtrace)</li>
<li>Present QC for raw read, alignment, and expression results (MultiQC)</li>
</ol>



<h4 id="running-nf-coresrnaseq">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/srna/#running-nf-coresrnaseq">running nf-core/sRNAseq</a>
</h4>
<p><img src="/sRNA/smRNAseq_log.png" alt=""></p>
<p>small RNA multiqc
<a href="../workshop-small-RNAseq_multiqc_report.html">link to small RNA multiqc</a></p>



<h1 id="downloading-the-data-to-you-local-pc">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/srna/#downloading-the-data-to-you-local-pc">Downloading the data to you local PC</a>
</h1>
<p>working on an hpc environment has some drawbacks, one being that viewing images and html files isn&rsquo;t as straightforward as it is on your pc. So that we can view the outputs together, this next step is to download the data from the nectar cloud to you pc</p>
<p>to promote be good practice, we will first compress the whole directory using <code>tar</code> and run an md5sum check, to show that the download hasn&rsquo;t corrupted the data.</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">cd</span> ~/workshop
</span></span><span class="line"><span class="cl">tar -cvf smallRNA.tar smallRNA
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl">md5sum smallRNA.tar &gt; smallRNA.md5sum
</span></span><span class="line"><span class="cl">md5sum smallRNA.tar</span></span></code></pre></div>
<p>the md5sum value should print to you screen, you could also save it as a txt file to view later</p>

  

<pre tabindex="0"><code>bee6e0161d40eb934c2ad0b4c2db1898  smallRNA.tar</code></pre>
<p>you can do the same on you downloaded copy, to make sure it&rsquo;s exactly the same data.</p>



<h3 id="downloading-with-scp">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/srna/#downloading-with-scp">downloading with scp</a>
</h3>
<p>we will be using <em><strong>scp</strong></em> (ssh copy)</p>
<p>You will need the same information as what you used to log in with <code>ssh</code></p>
<table>
  <thead>
      <tr>
          <th style="text-align: left"></th>
          <th style="text-align: left"></th>
          <th style="text-align: left"></th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td style="text-align: left">1</td>
          <td style="text-align: left">username</td>
          <td style="text-align: left"><strong>workshop</strong></td>
      </tr>
      <tr>
          <td style="text-align: left">2</td>
          <td style="text-align: left">password</td>
          <td style="text-align: left"><strong>Sagc_2024</strong></td>
      </tr>
      <tr>
          <td style="text-align: left">3</td>
          <td style="text-align: left">IP address</td>
          <td style="text-align: left"><em><strong>given on the day</strong></em></td>
      </tr>
  </tbody>
</table>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># open a local shell terminal, make a directory and move to it</span>
</span></span><span class="line"><span class="cl">mkdir -p  ~/workshop_RNAseq/smallRNA 
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> ~/workshop_RNAseq/smallRNA</span></span></code></pre></div>
<p>#example only, use your own IP address
IP=[yourIPaddress]</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">scp -r workshop@<span class="si">${</span><span class="nv">IP</span><span class="si">}</span>:/home/workshop/workshop/smallRNA.tar .
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#eg. scp -r workshop@${IP}:/home/workshop/workshop/smallRNA . </span></span></span></code></pre></div>
<ul>
<li>this file is 617M, it should take a minute or less to downlaod</li>
</ul>
<p>now on your local environment, you can run the same md5sum command</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">md5sum smallRNA.tar</span></span></code></pre></div>
<p>the value should be exactly the same, showing the files are exact to the bit.</p>
<p>you can now uncompress the directory</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">tar -xvf smallRNA.tar</span></span></code></pre></div>

  

  

  





  
  



      </div>
    </div>
  </main>
  
  <footer class="Footer">
    <div class="u-wrapper">
      <div class="u-padding u-noboosting">
        2024 Daniel Thomson &#32; <a href="https://sa-genomics.com.au/our-team" rel="license">SA Genomics Centre</a>
      </div>
    </div>
  </footer>

</body>

</html>
