<!DOCTYPE html>
<html lang="en">

<head><script src="/workshop_nfRNAseq/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=workshop_nfRNAseq/livereload" data-no-instant defer></script>
  
  <meta charset="utf-8">



<meta name="viewport" content="width=device-width, initial-scale=1.0">


  
    <meta name="description" content="
  Instructions to set up your personal computer


You will need to bring your own PC

We will be using the ARDC Nectar Research Cloud to perform RNAseq analyses





  Requirements


terminal applciation installed
or alternativelly Visual Studio Code installed
please see instructions below if don&rsquo;t have either of these





  Accessing nectar using a ssh (Secure Shell) command

To access Nectar you will need the following">
  


<meta name="color-scheme" content="light dark">







<meta name="generator" content="Hugo 0.135.0">
  <title>Setup | SAGG bioinformatics workshop</title>
  <link rel="canonical" href="http://localhost:1313/workshop_nfRNAseq/setup_environment/">




  








  
    
  
  
    <link rel="stylesheet" href="/workshop_nfRNAseq/css/base.css">

    <link rel="stylesheet" href="/workshop_nfRNAseq/css/syntax.css">



</head>

<body>
  <nav class="u-background">
  <div class="u-wrapper">
    <ul class="Banner">
      <li class="Banner-item Banner-item--title">
        <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/">SAGG bioinformatics workshop</a>
      </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/agenda/">agenda</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/setup_environment/">setup</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nextflow/">nextflow</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nf-core/">nf-core</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/nfrnaseq/">nfRNAseq</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/differentialabundance/">DifferentialAbundance</a>
        </li>
      
        
        
        <li class="Banner-item">
          <a class="Banner-link u-clickable" href="/workshop_nfRNAseq/srna/">sRNA</a>
        </li>
      
    </ul>
  </div>
</nav>

  <main>
    <div class="u-wrapper">
      <div class="u-padding">
        

  <header class="Heading">
  <h1 class="Heading-title">
    <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/setup_environment/" rel="bookmark">Setup</a>
  </h1>
  
</header>
  


<h3 id="instructions-to-set-up-your-personal-computer">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/setup_environment/#instructions-to-set-up-your-personal-computer">Instructions to set up your personal computer</a>
</h3>
<ul>
<li><em>You will need to bring your own PC</em></li>
</ul>
<p>We will be using the <a href="https://ardc.edu.au/services/ardc-nectar-research-cloud/">ARDC Nectar Research Cloud</a> to perform RNAseq analyses</p>
<p><img src="nectar_logo.png" alt=""></p>



<h3 id="requirements">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/setup_environment/#requirements">Requirements</a>
</h3>
<ul>
<li>terminal applciation installed</li>
<li>or alternativelly Visual Studio Code installed<br>
<em>please see instructions below if don&rsquo;t have either of these</em></li>
</ul>



<h3 id="accessing-nectar-using-a-ssh-secure-shell-command">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/setup_environment/#accessing-nectar-using-a-ssh-secure-shell-command">Accessing nectar using a ssh (Secure Shell) command</a>
</h3>
<p>To access <code>Nectar</code> you will need the following</p>
<table>
  <thead>
      <tr>
          <th style="text-align: left"></th>
          <th style="text-align: left"></th>
          <th style="text-align: left"></th>
      </tr>
  </thead>
  <tbody>
      <tr>
          <td style="text-align: left">1</td>
          <td style="text-align: left">username</td>
          <td style="text-align: left"><strong>workshop</strong></td>
      </tr>
      <tr>
          <td style="text-align: left">2</td>
          <td style="text-align: left">password</td>
          <td style="text-align: left"><strong>Sagc_2024</strong></td>
      </tr>
      <tr>
          <td style="text-align: left">3</td>
          <td style="text-align: left">IP address</td>
          <td style="text-align: left"><em><strong>given on the day</strong></em></td>
      </tr>
  </tbody>
</table>
<p>Your individual IP address will be given on the day. Each participant will have their own VM (virtual machine).
<strong>Do not log in with the same IP address as someone else!</strong>*</p>
<p>your ssh command should look like this, but with a different IP address</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">ssh workshop@172.243.264.45</span></span></code></pre></div>
<p>press enter, and you should see be prompted with some text</p>

  

<pre tabindex="0"><code>The authenticity of host &#39;118.138.235.141 (118.138.235.141)&#39; can&#39;t be established.
ECDSA key fingerprint is SHA256:dgHPXk3BkFPTErqpjAYkB5qAnHLYeUZE7qg3YKcWVQw.
Are you sure you want to continue connecting (yes/no/[fingerprint])?</code></pre>
<ul>
<li>type in &lsquo;yes&rsquo;</li>
<li>enter your password if prompted</li>
</ul>
<p>you should see your username in green text, e.g.</p>

  

<pre tabindex="0"><code>workshop@workshop</code></pre>
<p>This means you have logged into the virtual machine.</p>
<p>you can now navigate to the workshop materials, which have been loaded to your instance.</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">cd</span> ~/workshop/
</span></span><span class="line"><span class="cl">ls -l</span></span></code></pre></div>



<h2 id="install-and-set-up-a-terminal-application-if-needed">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/setup_environment/#install-and-set-up-a-terminal-application-if-needed">Install and set up a terminal application (if needed)</a>
</h2>



<h4 id="linux-terminals">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/setup_environment/#linux-terminals">Linux terminals</a>
</h4>
<p>If you use Linux, then likely no explanation is needed. Open your preferred terminal program</p>



<h4 id="os-x-mac">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/setup_environment/#os-x-mac">OS X (Mac)</a>
</h4>
<p>Mac operating systems have a terminal program, called Terminal. Just look for it in your Applications folder, or hit Command + Space and type ‘terminal’. You may find that other, 3rd party terminal programs are .</p>
<p>Use <code>ssh</code> in the <em>Terminal</em> app. For example, if your IP is <code>172.243.264.45</code> and user name is <code>user</code>, you would enter:</p>

  

<pre tabindex="0"><code>ssh user@172.243.264.45</code></pre>



<h3 id="windows">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/setup_environment/#windows">Windows</a>
</h3>
<p>The latest builds of Windows 10 and 11 come with SSH server and client. To see if your windows installation has SSH:</p>
<ul>
<li>start Command Prompt (cmd) or Powershell</li>
<li>type in &lsquo;ssh&rsquo;</li>
</ul>
<p>If SSH is available, you should see something like</p>

  

<pre tabindex="0"><code>usage: ssh [-46AaCfGgKkMNnqsTtVvXxYy] [-B bind_interface]
           [-b bind_address] [-c cipher_spec] [-D [bind_address:]port]
           [-E log_file] [-e escape_char] [-F configfile] [-I pkcs11]
           [-i identity_file] [-J [user@]host[:port]] [-L address]
           [-l login_name] [-m mac_spec] [-O ctl_cmd] [-o option] [-p port]
           [-Q query_option] [-R address] [-S ctl_path] [-W host:port]
           [-w local_tun[:remote_tun]] destination [command]</code></pre>
<p>If SSH is not available, please install <a href="https://www.putty.org/">PuTTY</a>.</p>
<p>Open PuTTY and connect using the IP address of your VM as the host name, and set port to 22.</p>
<p>When prompted, type in your username and password.</p>



<h1 id="environment-setup">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/setup_environment/#environment-setup">Environment Setup</a>
</h1>
<pre><code>**NO NEED TO CONTINUE** if accessing this content on the nectar instance
</code></pre>
<p>On the day of the workshop this will have already been done, and will be available when you log into the Nectar instance using <em>ssh</em> <br>
however if you wish to use your own computing environment or from scratch you will have to install the dependencies</p>
<ul>
<li>install nextflow</li>
<li>install conda</li>
<li>install nf-core tools</li>
<li>install singularity</li>
<li>install GO</li>
</ul>
<p>you can fistly check whether these have been installed with the following commands</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">nextflow -version
</span></span><span class="line"><span class="cl">conda -version
</span></span><span class="line"><span class="cl">nf-core -version
</span></span><span class="line"><span class="cl">singularity -version</span></span></code></pre></div>



<h4 id="install-nextflow">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/setup_environment/#install-nextflow">install nextflow</a>
</h4>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir ~/bin <span class="o">&amp;&amp;</span> <span class="nb">cd</span> ~/bin
</span></span><span class="line"><span class="cl">curl -s https://get.nextflow.io <span class="p">|</span> bash
</span></span><span class="line"><span class="cl">chmod +x nextflow</span></span></code></pre></div>
<p>adding ~/bin to PATH # in ~/.bashrc</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s1">&#39;export PATH=~/bin:${PATH}&#39;</span> &gt;&gt; ~/.bashrc <span class="o">&amp;&amp;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    <span class="nb">source</span> ~/.bashrc</span></span></code></pre></div>
<p>test installation</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">nextflow -version</span></span></code></pre></div>



<h4 id="setting-the-nextflow-singularity-cachedir">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/setup_environment/#setting-the-nextflow-singularity-cachedir">Setting the nextflow singularity cachedir</a>
</h4>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">mkdir -p ~/SingularityCacheDir
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#export NXF_SINGULARITY_CACHEDIR=~/SingularityCacheDir</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s1">&#39;export NXF_SINGULARITY_CACHEDIR=~/SingularityCacheDir&#39;</span> &gt;&gt; ~/.bashrc <span class="o">&amp;&amp;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    <span class="nb">source</span> ~/.bashrc
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># check cache dir</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="nv">$NXF_SINGULARITY_CACHEDIR</span></span></span></code></pre></div>



<h4 id="install-singularity">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/setup_environment/#install-singularity">install Singularity</a>
</h4>
<ul>
<li>install the following dependencies: (Ubuntu)</li>
</ul>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">sudo apt-get update <span class="o">&amp;&amp;</span> sudo apt-get install -y <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    build-essential <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    uuid-dev <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    libgpgme-dev <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    squashfs-tools <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    libseccomp-dev <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    wget <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    pkg-config <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    git <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    cryptsetup-bin</span></span></code></pre></div>
<p>if you are using an HPC environment without superuser (su) access, then you will have to ask IT/administrator to install it
Sahmri HPC, Nectar or Deepthought will already have it installed</p>
<ul>
<li>install GO</li>
<li>GO is a dependency of Singularity</li>
</ul>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">export</span> <span class="nv">VERSION</span><span class="o">=</span>1.23.2 <span class="nv">OS</span><span class="o">=</span>linux <span class="nv">ARCH</span><span class="o">=</span>amd64 <span class="o">&amp;&amp;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    wget https://dl.google.com/go/go<span class="nv">$VERSION</span>.<span class="nv">$OS</span>-<span class="nv">$ARCH</span>.tar.gz <span class="o">&amp;&amp;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    sudo tar -C /usr/local -xzvf go<span class="nv">$VERSION</span>.<span class="nv">$OS</span>-<span class="nv">$ARCH</span>.tar.gz <span class="o">&amp;&amp;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    rm go<span class="nv">$VERSION</span>.<span class="nv">$OS</span>-<span class="nv">$ARCH</span>.tar.gz
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># set up your environment for Go.</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s1">&#39;export GOPATH=${HOME}/go&#39;</span> &gt;&gt; ~/.bashrc <span class="o">&amp;&amp;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    <span class="nb">echo</span> <span class="s1">&#39;export PATH=/usr/local/go/bin:${PATH}:${GOPATH}/bin&#39;</span> &gt;&gt; ~/.bashrc <span class="o">&amp;&amp;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    <span class="nb">source</span> ~/.bashrc</span></span></code></pre></div>
<p>Install Singularity</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># download singularity</span>
</span></span><span class="line"><span class="cl"><span class="nb">export</span> <span class="nv">VERSION</span><span class="o">=</span>3.9.6 <span class="o">&amp;&amp;</span> <span class="c1"># adjust this as necessary \</span>
</span></span><span class="line"><span class="cl">    wget https://github.com/sylabs/singularity/releases/download/v<span class="si">${</span><span class="nv">VERSION</span><span class="si">}</span>/singularity-ce-<span class="si">${</span><span class="nv">VERSION</span><span class="si">}</span>.tar.gz <span class="o">&amp;&amp;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    tar -xzf singularity-ce-<span class="si">${</span><span class="nv">VERSION</span><span class="si">}</span>.tar.gz <span class="o">&amp;&amp;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    <span class="nb">cd</span> singularity-ce-<span class="si">${</span><span class="nv">VERSION</span><span class="si">}</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#compile Singularity</span>
</span></span><span class="line"><span class="cl">./mconfig <span class="o">&amp;&amp;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    make -C ./builddir <span class="o">&amp;&amp;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    sudo make -C ./builddir install
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># test installation</span>
</span></span><span class="line"><span class="cl">singularity  --version</span></span></code></pre></div>
<p>install Conda</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">cd</span> ~/bin <span class="o">&amp;&amp;</span> mkdir -p miniconda3
</span></span><span class="line"><span class="cl">wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda3/miniconda.sh
</span></span><span class="line"><span class="cl">bash miniconda3/miniconda.sh -b -u -p miniconda3
</span></span><span class="line"><span class="cl">rm miniconda3/miniconda.sh
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#### add PATH to ~/.bashrc</span>
</span></span><span class="line"><span class="cl"><span class="nb">echo</span> <span class="s1">&#39;export PATH=${PATH}:~/bin/miniconda3/bin&#39;</span> &gt;&gt; ~/.bashrc <span class="o">&amp;&amp;</span> <span class="se">\
</span></span></span><span class="line"><span class="cl"><span class="se"></span>    <span class="nb">source</span> ~/.bashrc</span></span></code></pre></div>
<p>add bioconda to conda channel</p>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl">conda config --add channels bioconda</span></span></code></pre></div>



<h4 id="install-nf-core-tools">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/setup_environment/#install-nf-core-tools">install nf-core tools</a>
</h4>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="c1"># chosing to install nf-core to a new conda environment called &#39;nf-core&#39;, with nexflow as well</span>
</span></span><span class="line"><span class="cl">conda create --name nf-core <span class="nv">python</span><span class="o">=</span>3.12 nf-core
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># initiate conda </span>
</span></span><span class="line"><span class="cl">conda init
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># if prompted close terminal and reopen</span>
</span></span><span class="line"><span class="cl"><span class="nb">exit</span>
</span></span><span class="line"><span class="cl"><span class="c1"># reopen with ssh command</span>
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># see whether the environment is there</span>
</span></span><span class="line"><span class="cl">conda env list
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1">#activate this environment</span>
</span></span><span class="line"><span class="cl">conda activate nf-core
</span></span><span class="line"><span class="cl">
</span></span><span class="line"><span class="cl"><span class="c1"># test installation</span>
</span></span><span class="line"><span class="cl">nextflow -v
</span></span><span class="line"><span class="cl">nf-core -v</span></span></code></pre></div>



<h1 id="samtools">
  <a class="Heading-link u-clickable" href="/workshop_nfRNAseq/setup_environment/#samtools">Samtools</a>
</h1>

  

<div class="highlight"><pre tabindex="0" class="chroma"><code class="language-bash" data-lang="bash"><span class="line"><span class="cl"><span class="nb">cd</span> ~/bin
</span></span><span class="line"><span class="cl">wget https://github.com/samtools/samtools/releases/download/1.21/samtools-1.21.tar.bz2
</span></span><span class="line"><span class="cl">tar -xvjf samtools-1.21.tar.bz2
</span></span><span class="line"><span class="cl">rm samtools-1.21.tar.bz2
</span></span><span class="line"><span class="cl"><span class="nb">cd</span> samtools-1.21/
</span></span><span class="line"><span class="cl">make
</span></span><span class="line"><span class="cl">make install</span></span></code></pre></div>
<p><a href="../Workshop_RNAseq_Intro.pdf">link to RNAseq Background slides</a></p>

  

  

  





  
  



      </div>
    </div>
  </main>
  
  <footer class="Footer">
    <div class="u-wrapper">
      <div class="u-padding u-noboosting">
        2024 Daniel Thomson &#32; <a href="https://sa-genomics.com.au/our-team" rel="license">SA Genomics Centre</a>
      </div>
    </div>
  </footer>

</body>

</html>
